server/src/main/java/com/octopus/teamcity/opentelemetry/server/endpoints/custom/CustomOTELEndpointHandler.java

```diff
@@ -52,6 +54,25 @@ public SetProjectConfigurationSettingsRequest GetSetProjectConfigurationSettings
         return new SetCustomProjectConfigurationSettingsRequest(request);
     }
 
+    @Override
+    public void mapParamsToModel(Map<String, String> params, Map<String, Object> model) {
+        model.put("otelEndpoint", params.get(PROPERTY_KEY_ENDPOINT));
+
+        var headers = new ArrayList<HeaderDto>();
+        params.forEach((k,v)->{
+            if (k.startsWith(PROPERTY_KEY_HEADERS)) {
+                var key = k.substring(PROPERTY_KEY_HEADERS.length());
+                key = key.substring(1, key.length() - 1);
+                var header = new HeaderDto(key,
+                        EncryptUtil.isScrambled(v) ? EncryptUtil.unscramble(v) : v,
+                        EncryptUtil.isScrambled(v) ? "password" : "plaintext");
```

Comment:
This seems wrong. I do understand that this is pre-existing code. So, ignore if it is doing what it is supposed to.

Reply:
Yeah, pre-existing, and works. 🙈